impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultDependencyResolver.java

```diff
@@ -68,6 +68,22 @@
 @Singleton
 public class DefaultDependencyResolver implements DependencyResolver {
 
+    /**
+     * Cache of information about the modules contained in a path element.
+     * This cache is created when first needed. It may be never created.
+     *
+     * <p><b>TODO:</b> This field should not be in this class, because the cache should be global to the session.
+     * This field exists here only temporarily, until clarified where to store session-wide caches.</p>
+     *
+     * @see #moduleCache()
+     */
+    private PathModularizationCache moduleCache;
```

Comment:
except of static this is the same outcome.
```suggestion
    private final static PathModularizationCache MODULE_CACHE = new PathModularizationCache();
```